python/paddle/fluid/contrib/tests/test_bf16_utils.py

```diff
@@ -0,0 +1,194 @@
+                          {'lstm'}, {'lstm'})
+
+    def test_amp_lists_8(self):
+        bf16_list = copy.copy(amp.bf16.amp_lists.bf16_list)
+        fp32_list = copy.copy(amp.bf16.amp_lists.fp32_list)
+        gray_list = copy.copy(amp.bf16.amp_lists.gray_list)
+
+        fp32_list.add('reshape2')
+        gray_list.remove('reshape2')
+
+        amp_lists_ = amp.AutoMixedPrecisionListsBF16(
+            custom_fp32_list={'reshape2'})
+        self.assertEqual(amp_lists_.bf16_list, bf16_list)
+        self.assertEqual(amp_lists_.fp32_list, fp32_list)
+        self.assertEqual(amp_lists_.gray_list, gray_list)
+
+    def test_amp_list_9(self):
+        bf16_list = copy.copy(amp.bf16.amp_lists.bf16_list)
+        fp32_list = copy.copy(amp.bf16.amp_lists.fp32_list)
+        gray_list = copy.copy(amp.bf16.amp_lists.gray_list)
+
+        bf16_list.add('reshape2')
+        gray_list.remove('reshape2')
+
+        amp_lists_ = amp.AutoMixedPrecisionListsBF16(
+            custom_bf16_list={'reshape2'})
+        self.assertEqual(amp_lists_.bf16_list, bf16_list)
+        self.assertEqual(amp_lists_.fp32_list, fp32_list)
+        self.assertEqual(amp_lists_.gray_list, gray_list)
+
+    def test_find_op_index(self):
+        block = fluid.default_main_program().global_block()
+        op_desc = core.OpDesc()
+        idx = amp.bf16.amp_utils.find_op_index(block.desc, op_desc)
+        assert (idx == -1)
+
+    def test_is_in_fp32_varnames(self):
+        block = fluid.default_main_program().global_block()
+
+        var1 = block.create_var(name="X", shape=[3], dtype='float32')
+        var2 = block.create_var(name="Y", shape=[3], dtype='float32')
+        var3 = block.create_var(name="Z", shape=[3], dtype='float32')
+        op1 = block.append_op(
+            type="abs", inputs={"X": [var1]}, outputs={"Out": [var2]})
+        op2 = block.append_op(
+            type="abs", inputs={"X": [var2]}, outputs={"Out": [var3]})
+        amp_lists_1 = amp.AutoMixedPrecisionListsBF16(
+            custom_fp32_varnames={'X'})
+        assert amp.bf16.amp_utils._is_in_fp32_varnames(op1, amp_lists_1)
+        amp_lists_2 = amp.AutoMixedPrecisionListsBF16(
+            custom_fp32_varnames={'Y'})
+        assert amp.bf16.amp_utils._is_in_fp32_varnames(op2, amp_lists_2)
+        assert amp.bf16.amp_utils._is_in_fp32_varnames(op1, amp_lists_2)
+
+    def test_find_true_post_op(self):
+
+        block = fluid.default_main_program().global_block()
+
+        var1 = block.create_var(name="X", shape=[3], dtype='float32')
+        var2 = block.create_var(name="Y", shape=[3], dtype='float32')
+        var3 = block.create_var(name="Z", shape=[3], dtype='float32')
+        op1 = block.append_op(
+            type="abs", inputs={"X": [var1]}, outputs={"Out": [var2]})
+        op2 = block.append_op(
+            type="abs", inputs={"X": [var2]}, outputs={"Out": [var3]})
+        res = amp.bf16.amp_utils.find_true_post_op(block.ops, op1, "Y")
```

Comment:
Similar comment as for `test_find_op_index`

Reply:
Please look at my answer about `find_op_index`